### PR TITLE
feat: add secrets generation, security ctx, container command/args and container resources

### PR DESCRIPTION
**Pull request by bjarneo**
- Removed ingress as we do not need it
- Refactor to javascript for the cdk8s and the github action
- Liveness/readiness is checking the tcp socket connection 
- Add secrets generation
- Add security context fixes
- Add container command / args
- Add container resources
- Use external secrets CRD to generate a class to be used for the secrets creation

Note: Do a squash merge

### CODE REVIEW

File: .gitattributes [hunk 1]
```diff
@@ -0,0 +1 @@
+dist/** -diff linguist-generated=true
```
Comment (bjarneo) [inline .gitattributes:1]: Add comment to what this actually do

File: README.md [hunk 2]
```diff
@@ -17,17 +17,21 @@ It aims to a generic generator that can generate any manifest we'll need for dep
...
     # These are optional
-    healthcheck_path: /healthz
     replicas: 1
     port: 80
+    container_size: medium
     container_port: 80
-    ingress: true
-    ingress_host: prevent.dev.dignio.dev
-    ingress_path: /
+    container_command: '["curl"]'
```
Comment (bjarneo) [inline README.md:30]: can we remove the single quotes?

File: README.md [hunk 3]
```diff
@@ -17,17 +17,21 @@ It aims to a generic generator that can generate any manifest we'll need for dep
     namespace: development
     docker_image: <org-id>.dkr.ecr.<region>.amazonaws.com/<repo-name>:<tag>
 
+    # Add secret config. Optional.
+    # Both has to be set.
+    secretsmanager: true
```
Comment (bjarneo) [inline README.md:22]: Describe how this works in the README

File: README.md [hunk 4]
```diff
@@ -39,15 +43,16 @@ The output from this action is the generated manifest. Can be accessed by other
...
-pipenv install
+```bash
+npm install
 ```
 
 ## Testing this action locally
 
 ```bash
-pipenv run generate
+npm run dev
```
Comment (bjarneo) [inline README.md:53]: mention hot reloading

File: action.yaml [hunk 5]
```diff
@@ -1,66 +1,54 @@
...
+  container_size:
+    description: The size of the container. CPU/Memory. small|medium|large|xlarge
+    required: false
+    default: ''
+  container_command:
+    description: The command to be passed to the docker container "['curl']"
+    required: false
+    default: ''
+  container_args:
+    description: The args to be passed to the docker container "['-I', 'https://www.google.com']"
```
Comment (bjarneo) [inline action.yaml:37]: Double check single quotes, if single can be used

Comment (bjarneo) [reply action.yaml:37]: use i.e or e.g

File: action.yaml [hunk 6]
```diff
@@ -1,66 +1,54 @@
...
     required: false
     default: 'false'
-  ingress_path:
-    description: 'The application root URL path'
-    required: false
-    default: '/'
-  ingress_host:
-    description: 'The host name of the application. I.e. prevent.dev.dignio.dev'
+  cluster_name:
+    description: The clustername
```
Comment (bjarneo) [inline action.yaml:49]: why? (cluster name)

Comment (bjarneo) [reply action.yaml:49]: example

File: cdk8s.yaml [hunk 7]
```diff
@@ -0,0 +1,4 @@
+language: typescript
```
Comment (bjarneo) [inline cdk8s.yaml:1]: Remove this as we do not use typescript

File: package.json [hunk 8]
```diff
@@ -0,0 +1,37 @@
+{
+  "name": "generate-manifest",
+  "version": "1.0.0",
+  "main": "main.js",
+  "type": "module",
+  "author": "Bjarne Oeverli",
+  "license": "MIT",
+  "private": true,
+  "scripts": {
+    "build": "ncc build main.js -o dist --source-map --license licenses.txt",
```
Comment (bjarneo) [inline package.json:10]: Add to readme if it is not there already

File: src/service/webservice.js [hunk 9]
```diff
@@ -0,0 +1,155 @@
...
+
+        // Information regarding security context
+        // https://kubernetes.io/docs/tasks/configure-pod-container/security-context/
+        securityContext: {
+            ensureNonRoot: true,
+            // https://hub.armo.cloud/docs/c-0017
+            // if we set this to true, it is not possible to write to /tmp
+            readOnlyRootFilesystem: false,
+            privileged: false,
+            user: 1000,
```
Comment (bjarneo) [inline src/service/webservice.js:71]: add hub armo docs

File: README.md (hunk 2 above)
Comment (bjarneo) [reply README.md:30]: no, because then YAML will treat it as an aray

**Review by lemonsaurus — APPROVED** [review]
I'm happy with the state of this refactor now.

This is a huge change, and it's hard to be 100% certain of anything, but since it's been carefully tested already and we've done a mob review of all the lines, I'm comfortable approving this for merge.

**Pull request merged**